From ccac9b1ea8ef85678473f6f6825e482ca4c19ebc Mon Sep 17 00:00:00 2001
From: Benoit Bovy <benbovy@gmail.com>
Date: Mon, 2 Dec 2024 15:52:58 +0100
Subject: [PATCH] fix openssl header not found in specific cases

E.g., on MacOS with s2geometry built from source and installed in a
custom directory and linked against openssl installed in an other,
standard directory.
---
 CMakeLists.txt | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 5fb3e93..6d040ad 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -173,6 +173,11 @@ elseif(${S2_SOURCE} STREQUAL "SYSTEM")
   endif()
 endif()

+# this might be needed since s2geometry includes it in general
+# but not for any target explicilty?
+find_package(OpenSSL REQUIRED)
+target_include_directories(${s2_NOALIAS_TARGET} INTERFACE ${OPENSSL_INCLUDE_DIR})
+
 # --- Abseil (bundled build not supported)

 find_package(absl REQUIRED)
--
2.36.0
